objects/src/accounts/delta/mod.rs

```diff
@@ -114,10 +127,18 @@ impl Deserializable for AccountDelta {
 /// - Storage or vault were updated, but the nonce was either not updated or set to 0.
 /// - Storage and vault were not updated, but the nonce was updated.
 fn validate_nonce(
+    is_new_account: bool,
     nonce: Option<Felt>,
     storage: &AccountStorageDelta,
     vault: &AccountVaultDelta,
 ) -> Result<(), AccountDeltaError> {
+    if is_new_account {
+        if nonce != Some(ZERO) {
+            return Err(AccountDeltaError::IncorrectNonceForNewAccount(nonce));
+        }
```

Comment:
I'm not sure this is correct: nonce for a new account would start out as zero, but after the update it would not be zero. So, the nonce in the delta would not be zero. We could potentially force the nonce for the first update to be one - but I'm not sure that's a good idea.